TASVideos/WikiModules/PublicationsByPlatform.cshtml.cs

```diff
@@ -5,11 +5,18 @@ namespace TASVideos.WikiModules;
 [WikiModule(ModuleNames.PublicationsByPlatform)]
 public class PublicationsByPlatform(IGameSystemService platforms) : WikiViewComponent
 {
-	public IReadOnlyList<SystemsResponse> Platforms { get; private set; } = null!;
+	public IReadOnlyList<(string DisplayName, string Code)> Platforms { get; private set; } = null!;
 
-	public async Task<IViewComponentResult> InvokeAsync()
+	public async Task<IViewComponentResult> InvokeAsync(IList<string> systemCodes)
 	{
-		Platforms = (await platforms.GetAll()).ToArray();
+		var extant = await platforms.GetAll();
+		Platforms = systemCodes.Select(s => s.Split('-').Select(id => extant.FirstOrDefault(sys => sys.Code.Equals(id, StringComparison.OrdinalIgnoreCase))).ToArray())
```

Comment:
Does this add grouping of some of the systems in the whole table, or does it just narrow down the table to only show the systems passed explicitly?

Reply:
The list is drawn from the module's argument now, and the full list from `IGameSystemService` is only used to sanitise it.

Reply:
The point of the request was to automatically show all platforms with pubs without having to manually add them every time a new platform is published. Reducing displayed systems to only those explicitly passed defeats the purpose of the module.